src/base/filesystemwatcher.cpp

```diff
@@ -167,113 +150,36 @@ void FileSystemWatcher::processPartialTorrents()
-            qDebug("Partial torrent detected at: %s", qUtf8Printable(fileAbsPath));
-            qDebug("Delay the file's processing...");
-            m_partialTorrents.insert(fileAbsPath, 0);
-        }
+        else if (!m_partialTorrents.contains(fileAbsPath))
+            m_partialTorrents[fileAbsPath] = 0;
     }
 
-    if (!m_partialTorrents.empty())
-        startPartialTorrentTimer();
-}
-
-#if !defined Q_OS_WIN && !defined Q_OS_HAIKU
-bool FileSystemWatcher::isNetworkFileSystem(QString path)
-{
-    QString file = path;
-    if (!file.endsWith("/"))
-        file += "/";
-    file += ".";
-    struct statfs buf;
-    if (!statfs(file.toLocal8Bit().constData(), &buf)) {
-#ifdef Q_OS_MAC
-        // XXX: should we make sure HAVE_STRUCT_FSSTAT_F_FSTYPENAME is defined?
-        return ((strcmp(buf.f_fstypename, "nfs") == 0) || (strcmp(buf.f_fstypename, "cifs") == 0) || (strcmp(buf.f_fstypename, "smbfs") == 0));
-#else
-        return ((buf.f_type == static_cast<long>(CIFS_MAGIC_NUMBER))
-            || (buf.f_type == static_cast<long>(NFS_SUPER_MAGIC))
-            || (buf.f_type == static_cast<long>(SMB_SUPER_MAGIC)));
-#endif
-    }
-    else {
-        std::cerr << "Error: statfs() call failed for " << qPrintable(file) << ". Supposing it is a local folder..." << std::endl;
-        switch(errno) {
-        case EACCES:
-            std::cerr << "Search permission is denied for a component of the path prefix of the path" << std::endl;
-            break;
-        case EFAULT:
-            std::cerr << "Buf or path points to an invalid address" << std::endl;
-            break;
-        case EINTR:
-            std::cerr << "This call was interrupted by a signal" << std::endl;
-            break;
-        case EIO:
-            std::cerr << "I/O Error" << std::endl;
-            break;
-        case ELOOP:
-            std::cerr << "Too many symlinks" << std::endl;
-            break;
-        case ENAMETOOLONG:
-            std::cerr << "path is too long" << std::endl;
-            break;
-        case ENOENT:
-            std::cerr << "The file referred by path does not exist" << std::endl;
-            break;
-        case ENOMEM:
-            std::cerr << "Insufficient kernel memory" << std::endl;
-            break;
-        case ENOSYS:
-            std::cerr << "The file system does not detect this call" << std::endl;
-            break;
-        case ENOTDIR:
-            std::cerr << "A component of the path is not a directory" << std::endl;
-            break;
-        case EOVERFLOW:
-            std::cerr << "Some values were too large to be represented in the struct" << std::endl;
-            break;
-        default:
-            std::cerr << "Unknown error" << std::endl;
-        }
+    if (!torrents.empty())
+        emit torrentsAdded(torrents);
 
-        std::cerr << "Errno: " << errno << std::endl;
-        return false;
-    }
+    if (!m_partialTorrents.empty())
```

Comment:
Isn't it possible that m_partialTorrentTimer is already active at this point?

Reply:
it makes sense to check `m_partialTorrentTimer` status before `start()`